tools/e2e-agent/zfs.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"k8s.io/klog/v2"
+)
+
+type Zpool struct {
+	PoolDiskPath string `json:"poolDiskPath"`
+	PoolName     string `json:"poolName"`
+}
+
+// ZfsVersion check zfs version installed on node
+func ZfsVersion(w http.ResponseWriter, r *http.Request) {
+	var msg string
+	klog.Info("Get zfs version installed")
+
+	zfsVersionCommand := "dmesg | grep ZFS"
```

Comment:
I'm not familiar with zfs - however using grep on dmesg can yield multiple hits of which only one may be relevant
Not 100% sure that this would work  but `modinfo --field=version zfs`  should yield the desired value.

Reply:
[fixed](https://github.com/openebs/openebs-e2e/compare/b4c7aae700e90482dc2e8adf8dd674a6b2b599ad..64e983b7d7e46bca97e9e503bd322affe0e124d0)